Better UI for upgrading an OPAM 1.2.2 root when bwrap missing

This transcript is not great:

```
dra@bionic:~$ opam/opam update
This development version of opam requires an update to the layout of /home/dra/.opam from version 1.2 to version 2.0, which can't be reverted.
You may want to back it up before going further.
Type "yes" to perform the update and continue: yes
Format upgrade done.

-_-_ Rerunning init and update -_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
Checking for available remotes: rsync and local, git, mercurial, darcs. Perfect!
[ERROR] Missing dependencies -- the following commands are required for opam to operate:
  - bwrap: Sandboxing tool bwrap was not found. You should install 'bubblewrap'. See http://opam.ocaml.org/doc/2.0/FAQ.html#Why-opam-asks-me-to-install-bwrap.

-_-_ Updating repositories -_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
[default] synchronised from https://opam.ocaml.org
[NOTE] The repository 'default' will be *permanently* redirected to https://opam.ocaml.org/2.0 (opam-version >= "2.0~")
[default] Initialised
Update done, please now retry your command.
dra@bionic:~$ opam/opam update

-_-_ Updating package repositories -_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
[default] no changes from https://opam.ocaml.org/2.0
Now run 'opam upgrade' to apply any package updates.
dra@bionic:~$ opam/opam upgrade
The following actions will be performed:
  ∗ install ocaml 4.06.1 
Do you want to continue? [Y/n] y

-_-_ Gathering sources -_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-

-_-_ Processing actions _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
[ERROR] The compilation of ocaml failed at "/home/dra/.opam/opam-init/hooks/sandbox.sh build ocaml unix.cma gen_ocaml_config.ml".

#=== ERROR while compiling ocaml.4.06.1 =======================================#
# context      2.0.0~rc | linux/x86_64 | base-bigarray.base base-threads.base base-unix.base ocaml-base-compiler.4.06.1 | https://opam.ocaml.org/2.0#56bc0dc1
# path         ~/.opam/4.06.1/.opam-switch/build/ocaml.4.06.1
# command      ~/.opam/opam-init/hooks/sandbox.sh build ocaml unix.cma gen_ocaml_config.ml
# exit-code    1
# env-file     ~/.opam/log/ocaml-19965-ccabea.env
# output-file  ~/.opam/log/ocaml-19965-ccabea.out
### output ###
# The 'bwrap' command was not found. Install 'bubblewrap' on your system, or
# /home/dra/.opam/opam-init/hooks/sandbox.sh: line 5: OPAMROOT: unbound variable



-_-_ Error report _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
┌─ The following actions failed
│ λ build ocaml 4.06.1
└─ 
╶─ No changes have been performed
dra@bionic:~$ vim c^C
dra@bionic:~$ vim .opam/config
dra@bionic:~$ 
```

I don't think that when **upgrading** an already-inited root that opam should be displaying `ERROR` when the process succeeds/isn't aborted. It should be a warning in this case. Note that error in `sandbox.sh` in the `opam upgrade`. It seems that failed sandboxing needs to be accompanied with some kind of flag which causes other commands to check that part of init every time `opam` is run, rather than just proceeding and throwing errors later.

Alternatively, the init checks should be done **before** the root is upgraded (a dry-run). Then the upgrade could actually display an `ERROR` and leave an OPAM 1.2.2 root behind.

--- a/src/client/opamClient.ml
+++ b/src/client/opamClient.ml
@@ -619,12 +619,14 @@ let init_checks ?(hard_fail_exn=true) init_config =
   let required_deps =
     filter_tools (OpamFile.InitConfig.required_tools init_config)
   in
-  let hard_fail = check_tool
-      (fun s -> OpamConsole.error
-          "Missing dependencies -- \
-           the following commands are required for opam to operate:";
-        OpamConsole.errmsg "%s" s)
-      required_deps in
+  let hard_fail =
+    let msg = if hard_fail_exn then OpamConsole.error else OpamConsole.warning in
+    check_tool (fun s -> msg
+                   "Missing dependencies -- \
+                    the following commands are required for opam to operate:";
+                 OpamConsole.errmsg "%s" s)
+      required_deps
+  in
 
   if hard_fail && hard_fail_exn then OpamStd.Sys.exit_because `Configuration_error
   else not (soft_fail || hard_fail)

--- a/src/state/bwrap.sh
+++ b/src/state/bwrap.sh
@@ -1,8 +1,13 @@
 #!/bin/bash -ue
 
+ROOT="~/.opam"
+if [ -v OPAMROOT ]; then
+	ROOT=$OPAMROOT
+fi
+
 if ! command -v bwrap >/dev/null; then
     echo "The 'bwrap' command was not found. Install 'bubblewrap' on your system, or" >&2
-    echo "disable sandboxing in $OPAMROOT/config at your own risk." >&2
+    echo "disable sandboxing in $ROOT/config at your own risk." >&2
     echo "See https://github.com/projectatomic/bubblewrap for bwrap details." >&2
     echo "For 'bwrap' use in opam, see the FAQ:" >&2
     echo "  https://opam.ocaml.org/doc/2.0/FAQ.html#Why-does-opam-require-bwrap." >&2

